### PR TITLE
DAC API Interface

### PR DESCRIPTION
**Pull request by RadConMannBor**
Simple API interface for the DAC

### CODE REVIEW

File: firmware/library/L1_Drivers/dac.hpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <cstdint>
+#include <cstdio>
```
Comment (kammce) [inline firmware/library/L1_Drivers/dac.hpp]: It doesn't look like you are using cstdio. Therefore, you should remove this include.

File: firmware/library/L1_Drivers/dac.hpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+#include "L0_LowLevel/LPC40xx.h"
+#include "L1_Drivers/pin_configure.hpp"
```
Comment (kammce) [inline firmware/library/L1_Drivers/dac.hpp]: Ditto here.

File: firmware/library/L1_Drivers/dac.hpp [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <cstdint>
+#include <cstdio>
+#include "L0_LowLevel/LPC40xx.h"
```
Comment (kammce) [inline firmware/library/L1_Drivers/dac.hpp]: It doesn't look like you are using this include either. Please remove it.

File: firmware/library/L1_Drivers/dac.hpp [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+#include "L1_Drivers/pin_configure.hpp"
+
+class DAC
+{
+ public:
+    DAC(void);
+    virtual bool Init();
+    virtual bool WriteDAC(uint16_t dac_output);
```
Comment (kammce) [inline firmware/library/L1_Drivers/dac.hpp]: I think it would be great to have a SetVoltage(float) method for this as well.

**Comment by kammce** [comment]
Please only commit 1 commit.

**Comment by RadConMannBor** [comment]
Is there a way for me to remove the 4 other commits?    They are from earlier when I was trying to update my repository.  Do i just need to make a new pull request and try not to add them?  Or do i need to back my repository up before them and then update it?

File: firmware/library/L1_Drivers/dac.hpp [hunk 5]
```diff
@@ -11,6 +8,7 @@ class DAC
     DAC(void);
     virtual bool Init();
```
Comment (kammce) [inline firmware/library/L1_Drivers/dac.hpp]: Prefer to spell this out fully. Should be `Initialize()`

File: firmware/library/L1_Drivers/dac.hpp [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    virtual bool Initialize();
+    virtual bool WriteDac(uint16_t dac_output);
+    virtual bool SetVoltage(float voltage);
+    // Sine Demonstration Function
+    virtual void SineDac(uint32_t frequency_hz);
+    // Sawtooth Demonstration Function
+    virtual void SawtoothDac(uint32_t frequency_hz);
+    // Triangle Demonstration Function
+    virtual void TriangleDac(uint32_t frequency_hz);
+    virtual void SetBIAS(bool bias_level);
```
Comment (kammce) [inline firmware/library/L1_Drivers/dac.hpp]: Please follow Google naming convention.

File: firmware/library/L1_Drivers/dac.hpp [hunk 7]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "L1_Drivers/pin_configure.hpp"
```
Comment (kammce) [inline firmware/library/L1_Drivers/dac.hpp]: Remove this line.

**Pull request merged**